### PR TITLE
29/split up entry point

### PR DESCRIPTION
**Pull request by WickyNilliams**
### What is the context of this PR?
This change simplifies the entry point (`src/index.js`), by moving code around to increase cohesion (related code lives together). It also removes some indirection around configuring routing. All routes are now clearly visible in `App` component

### How to review 
Check changes are sensible. Ensure app works as before

### CODE REVIEW

**Comment by WickyNilliams** [comment]
[Split up entry point JS module (2)](https://trello.com/c/hlLeAXkT/29-split-up-entry-point-js-module-2)

**Comment by WickyNilliams** [comment]
Weirdly, codecov was failing on first push. So i rebased (just in case), pushed again, and now there's no codecov at all??!

File: src/index.js [hunk 1]
```diff
@@ -1,76 +1,32 @@
...
-
-import thunk from "redux-thunk";
-
-import rootReducer from "reducers"; // Or wherever you keep your reducers
-
+import { configureStore, configurePersistedStore } from "store/configureStore";
 import App from "containers/App";
 
-let useStorage = false;
+let useStorage = true;
```
Comment (hamishtaplin) [inline src/index.js]: A good time to improve this perhaps?

File: src/index.js [hunk 2]
```diff
@@ -1,76 +1,32 @@
...
-};
-
-useStorage
-  ? load(store).then(newState => {
-    render(App);
-  })
-  : render(App);
-
 if (module.hot) {
+  module.hot.accept("reducers", () => {
```
Comment (hamishtaplin) [inline src/index.js]: out of interest, was there a reason to swap these around?

File: src/index.js (hunk 1 above)
Comment (WickyNilliams) [inline src/index.js]: Crap, should have left that as false. What kind of improvements are you thinking?

File: src/index.js (hunk 2 above)
Comment (WickyNilliams) [inline src/index.js]: Accident! Will revert

File: src/index.js (hunk 1 above)
Comment (hamishtaplin) [inline src/index.js]: Dunno, only think I can think of is a query param?

File: src/index.js (hunk 2 above)
Comment (hamishtaplin) [inline src/index.js]: Doesn't matter, just wondering if there was a reason

**Pull request merged**